src/Npgsql/Compatibility/CrateDB/CrateDBDatabaseInfo.cs

```diff
@@ -0,0 +1,146 @@
+        // pg_proc- or pg_range-table.
+        static string GenerateCrateTypesQuery(bool forBaseTypes)
+        {
+            return
+$@"select 
+    'pg_catalog' as nspname, typname, oid, 0 as typrelid, 0 as typbasetype, 
+    '{(forBaseTypes ? "b" : "a")}' as type, 
+    {(forBaseTypes ? "0" : "typelem")} as elemoid
+from
+    pg_catalog.pg_type
+where
+    {(forBaseTypes ? "" : "not ")}typelem = 0";
+        }
+
+        /// <summary>
+        /// Overrides DatabaseInfo.LoadBackendTypes to use type loading queries adapted for CrateDB. 
+        /// </summary>
+        /// <param name="connector"></param>
+        /// <param name="timeout"></param>
+        /// <param name="async"></param>
+        /// <returns></returns>
+        protected override async Task LoadBackendTypes(NpgsqlConnector connector, NpgsqlTimeout timeout, bool async)
+        {
+            await LoadBackendTypes(connector, timeout, async, CrateTypesQueryForBaseTypes);
+            await LoadBackendTypes(connector, timeout, async, CrateTypesQueryForArrayTypes);
+        }
+
+        async Task LoadBackendTypes(NpgsqlConnector connector, NpgsqlTimeout timeout, bool async, string typesQuery)
+        {
+            var commandTimeout = 0;  // Default to infinity
+            if (timeout.IsSet)
+            {
+                commandTimeout = (int)timeout.TimeLeft.TotalSeconds;
+                if (commandTimeout <= 0)
+                    throw new TimeoutException();
+            }
+
+            using (var command = new NpgsqlCommand(typesQuery, connector.Connection))
+            {
+                command.CommandTimeout = commandTimeout;
+                command.AllResultTypesAreUnknown = true;
+                using (var reader = async ? await command.ExecuteReaderAsync() : command.ExecuteReader())
+                {
+                    while (async ? await reader.ReadAsync() : reader.Read())
+                    {
+                        timeout.Check();
+                        LoadBackendType(reader, connector);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Overrides DatabaseInfo.TryGetComposite as CrateDB does not support named composite types.
+        /// </summary>
+        /// <param name="pgName"></param>
+        /// <param name="connection"></param>
+        /// <param name="compositeType"></param>
+        /// <returns></returns>
+        internal override bool TryGetComposite(string pgName, NpgsqlConnection connection, out PostgresCompositeType compositeType)
+        {
+            compositeType = null;
+            return false;
+        }
+
+        /// <summary>
+        /// Overrides DatabaseInfo.AddVendorSpecificTypeMappings to adapt 
+        /// the type mappings to the types supported by CrateDB.
+        /// </summary>
+        /// <param name="mapper"></param>
+        internal override void AddVendorSpecificTypeMappings(INpgsqlTypeMapper mapper)
```

Comment:
This should probably be done as part of the new plugin API, rather than an additional `AddVendorSpecificTypeMappings` which won't be used anywhere else. Again, I'll think a little bit on architecture.